### ISSUE TITLE
make distcheck shows files that are uninstalled after run

### ISSUE THREAD

**Issue by tijuca** [obgm/libcoap#305]
I noticed that `make distcheck` shows some file removing issue due some non removed man pages. Happens even in recent tagged version 4.2.0. Didn't we have Travis check for `make distcheck`?

```
make[1]: Entering directory '/home/carsten/gitprojects/libcoap/libcoap-4.2.0/_build/sub'
ERROR: files left after uninstall:
./share/man/man3/coap_endpoint_set_default_mtu.3
./share/man/man3/coap_show_tls_version.3
./share/man/man3/coap_free_endpoint.3
./share/man/man3/coap_add_optlist_pdu.3
./share/man/man3/coap_delete_observer.3
./share/man/man3/coap_session_set_max_retransmit.3
./share/man/man3/coap_session_set_ack_random_factor.3
./share/man/man3/coap_session_str.3
./share/man/man3/coap_add_data.3
./share/man/man3/coap_resource_init.3
./share/man/man3/coap_new_endpoint.3
./share/man/man3/coap_add_data_blocked_response.3
./share/man/man3/coap_find_observer.3
./share/man/man3/coap_resource_set_mode.3
./share/man/man3/coap_register_response_handler.3
./share/man/man3/coap_new_client_session_psk.3
./share/man/man3/coap_resource_get_uri_path.3
./share/man/man3/coap_insert_optlist.3
./share/man/man3/coap_session_get_app_data.3
./share/man/man3/coap_delete_optlist.3
./share/man/man3/coap_new_context.3
./share/man/man3/coap_session_get_ack_random_factor.3
./share/man/man3/coap_new_optlist.3
./share/man/man3/coap_delete_observers.3
./share/man/man3/coap_session_max_pdu_size.3
./share/man/man3/coap_split_path.3
./share/man/man3/coap_register_ping_handler.3
./share/man/man3/coap_debug_set_packet_loss.3
./share/man/man3/coap_register_pong_handler.3
./share/man/man3/coap_delete_resource.3
./share/man/man3/coap_new_client_session.3
./share/man/man3/coap_get_log_level.3
./share/man/man3/coap_session_set_mtu.3
./share/man/man3/coap_resource_unknown_init.3
./share/man/man3/coap_session_get_max_transmit.3
./share/man/man3/coap_session_get_ack_timeout.3
./share/man/man3/coap_log.3
./share/man/man3/coap_resource_notify_observers.3
./share/man/man3/coap_find_attr.3
./share/man/man3/coap_dtls_is_supported.3
./share/man/man3/coap_add_token.3
./share/man/man3/coap_package_name.3
./share/man/man3/coap_encode_var_bytes.3
./share/man/man3/coap_session_set_app_data.3
./share/man/man3/coap_register_event_handler.3
./share/man/man3/coap_add_resource.3
./share/man/man3/coap_session_set_ack_timeout.3
./share/man/man3/coap_get_tls_library_version.3
./share/man/man3/coap_set_log_level.3
./share/man/man3/coap_context_set_pki.3
./share/man/man3/coap_dtls_pki_t.3
./share/man/man3/coap_add_option.3
./share/man/man3/coap_endpoint_str.3
./share/man/man3/coap_delete_all_resources.3
./share/man/man3/coap_set_log_handler.3
./share/man/man3/coap_register_handler.3
./share/man/man3/coap_resource_set_get_observable.3
./share/man/man3/coap_session_reference.3
./share/man/man3/coap_register_nack_handler.3
./share/man/man3/coap_context_set_pki_root_cas.3
./share/man/man3/coap_show_pdu.3
./share/man/man3/coap_context_set_psk.3
./share/man/man3/coap_free_context.3
./share/man/man3/coap_string_tls_version.3
./share/man/man3/coap_split_query.3
./share/man/man3/coap_add_attr.3
./share/man/man3/coap_tls_is_supported.3
./share/man/man3/coap_package_version.3
./share/man/man3/coap_new_client_session_pki.3
./share/man/man3/coap_pdu_init.3
./share/man/man3/coap_session_release.3
./share/man/man3/coap_set_show_pdu_output.3
make[1]: *** [Makefile:1319: distuninstallcheck] Error 1
```

I've currently no clue why this happen ... :cry: A wild guess is due the dynamic (and some complexity) we've introduced into the creation of the man pages. We need to add the files to the variable `man3_MANS` in `man/Makefile.am` so the files will honored while the unistallation.

**Comment by obgm** [comment]
There is a travis check for `make dist`.
`make distcheck` not so much — what is that supposed to do?

**Comment by tijuca** [comment]
This target does in principle what the name let assume you. It checks if the source is build able, installable but also removable by `make uninstall`. And exactly that step is failing. Not at RC issue but not nice. If someone has a hint how to extend man3_MANS than this issue is easy solvable.

**Comment by mrdeep1** [comment]
There are 2 things here that go towards making up the extra .3 files which are not cleared up by `make distcheck`

a2x (actually asciidoctor) creates different .3 files based on the entries in the NAME section (comma separated, up to the <space>-<space> summary).  It supports up to 10 different command/function name entries.

Secondly,  if there are more than 10 entries, then install-man in man/Makefile.am creates the addional entries.

The additional (over 10) entries are easy to separately define, the initial 10 entries would require parsing of the .txt files to determine what the alternative name entries are, and unfortunately automake does not appear to do this automatically.

A simple fix could be (but I am not that happy with deleting all the coap_*.3 files in the man3 directory n case there is a coap_ name clash)
````
diff --git a/man/Makefile.am b/man/Makefile.am
index f469543..42102ad 100644
--- a/man/Makefile.am
+++ b/man/Makefile.am
@@ -52,7 +52,13 @@ man7_MANS = $(MAN7)
        $(A2X) --doctype manpage --format manpage $<
        $(A2X) --doctype manpage --format xhtml $<
 
-# We are limited to 10 names, but synopsis can cover more
+# a2x builds alternative .3 files up to a limit of 10 names from the
+# NAME section, so that 'man' works against the alternative different
+# function names.
+# However, if there are more alternative names, they need to be defined
+# as per below.
+# Then all the aalternative names as well as the extras defined below need
+# to be cleaned up in a 'make unistall'.
 install-man: install-man3 install-man5 install-man7
        @echo ".so man3/coap_context.3" > coap_context_set_pki_root_cas.3
        @echo ".so man3/coap_pdu_setup.3" > coap_add_data_blocked_response.3
@@ -63,7 +69,13 @@ install-man: install-man3 install-man5 install-man7
        @echo ".so man3/coap_session.3" > coap_session_set_app_data.3
        @echo ".so man3/coap_logging.3" > coap_endpoint_str.3
        @echo ".so man3/coap_logging.3" > coap_session_str.3
-       $(INSTALL_DATA) *.3 "$(DESTDIR)$(man3dir)"
+       $(INSTALL_DATA) coap_*.3 "$(DESTDIR)$(man3dir)"
+
+# As well as removing the base 'man' pages, remove other .3 files built by
+# a2x, as well as install-man specials.  There is however an assumption that
+# all the coap_*(3) pages are not owned by anyone else.
+uninstall-man: uninstall-man3 uninstall-man5 uninstall-man7
+       -rm -f $(DESTDIR)$(man3dir)/coap_*.3
 
 CLEANFILES = *.3 *.5 *.7 *.xml *.html docbook-xsl.css
 
````
However there are additional `make distcheck` errors once past this point that also need to be fixed (complaints about gnutls even though only build for openssl at this point) which I am looking into.

**Comment by obgm** [comment]
Reading function names from the .txt files is not that difficult using sed:

```sh
#! /usr/bin/env sh

# This script extracts the names of libcoap functions that are
# referenced in the NAME section of a manual page txt source.
#
# The a2x input is read from stdin, and extracted function names
# are written to stdout.

sed -ne '/NAME/{
    :loop
     /coap_/{ s/ *, */\n/g; p }
    n
    /SYNOPSIS/q
    b loop
}' | sed -ne 's/^\(coap_[a-zA-Z_]\+\).*$/\1/p'
```

**Comment by mrdeep1** [comment]
@obgm - Thanks - in principle this works, I have got an equivalent of it into the Makefile.am in #307

**Comment by obgm** [comment]
@mrdeep1 Nice! (I ran into Makefile special character quoting hell with my approach—yours is much better).

**Comment by mrdeep1** [comment]
@tijuca Can this be closed now that #307 has been merged?

**Comment by tijuca** [comment]
Sure, let's close this issue then.
